### PR TITLE
Address most of Marie-Jose's comments

### PR DESCRIPTION
**Pull request by brunorijsman**

### CODE REVIEW

**Comment by Wojtek242** [comment]
Thanks a lot @brunorijsman! I haven't looked at the content changes yet, but one of your commits (I believe the second one) completely reformatted the indentation of the entire xml file which I'd rather not do as my editor will probably insist on undoing all of that. Would you be able to revert the indentation changes for this PR?

**Comment by brunorijsman** [comment]
Urghh… I will undo that.  I thought I had turned off Visual Studio Code’s “auto format on save” but evidently not.

**Comment by brunorijsman** [comment]
Fixed it.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 1]
```diff
@@ -217,13 +217,14 @@
         keywords will be used for the search engine. -->
 
    <abstract>
-     <t>The vision of a quantum internet is to fundamentally enhance Internet
+     <t>The vision of a quantum Internet is to enhance existing Internet
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I would not capitalize "internet" here since it's "**a** quantum internet"

File: draft-irtf-qirg-principles-09-rc.xml [hunk 2]
```diff
@@ -217,13 +217,14 @@
...
      technology by enabling quantum communication between any two points on
      Earth. To achieve this goal, a quantum network stack should be built from
      the ground up to account for the fundamentally new properties of quantum
-     entanglement. The first realisations of quantum entanglement networks are
-     imminent, but there is no practical proposal for how to organise, utilise,
-     and manage such networks. In this memo, we attempt to lay down the
+     entanglement. The first quantum entanglement networks have been realised
+     <xref target="Pompili21" />, but there is no practical proposal for how to
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:225]: I'm wondering whether this should also be included: https://www.science.org/doi/10.1126/sciadv.aba0959. It's also entanglement based, but without quantum memories or swapping.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 3]
```diff
@@ -245,9 +246,11 @@
      quantum computers of the future. A quantum network is not meant
      to replace classical networks, but rather form an overall hybrid
      classical-quantum network supporting new capabilities which are
-     otherwise impossible to realise <xref target="VanMeterBook" />.</t>
+     otherwise impossible to realise <xref target="VanMeterBook" />,
+     such as clustered or distributed quantum computing, quantum 
+     secure communications, and distributed quantum sensing.</t>
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I'd also add secure delegated quantum computing (i.e. blind quantum computing).

File: draft-irtf-qirg-principles-09-rc.xml [hunk 4]
```diff
@@ -260,23 +263,25 @@
      distances <xref target="Elliott03" /> <xref target="Peev09" /> <xref
      target="Aguado19" />.</t>
 
-     <t>Fully quantum networks capable of transmitting and managing entangled
-     quantum states in order to send, receive, and manipulate distributed
-     quantum information are now imminent <xref target="Castelvecchi18" /> <xref
-     target="Wehner18" />. Whilst a lot of effort has gone into physically
+     <t>The first fully quantum networks capable of transmitting and managing
+     entangled quantum states (defined in <xref target="entanglement" />) in order
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I think the better solution is to just avoid mentioning entanglement or quantum states until they are defined.

I propose to rephrase this sentence as follows: "The first fully quantum networks capable of transmitting and manipulating distributed quantum information ("prepare and measure" stage in [WEHNER18]) via an intermediate quantum node have recently been realized"

File: draft-irtf-qirg-principles-09-rc.xml [hunk 5]
```diff
@@ -260,23 +263,25 @@
...
+     to send, receive, and manipulate distributed quantum information have 
+     recently been realized <xref target="Pompili21" />
+     <xref target="Castelvecchi18" /> <xref target="Wehner18" />.
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: The Castelvecchi and Wehner references need to be moved or removed. I'd say remove both.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 6]
```diff
@@ -260,23 +263,25 @@
...
+     <xref target="Castelvecchi18" /> <xref target="Wehner18" />. 
+     Whilst a lot of effort has gone into physically
      realising and connecting such devices <xref target="Hensen15" />, and
      making improvements to their speed and error tolerance, there are no
      worked out proposals for how to run these networks. To draw an analogy
      with a classical network, we are at a stage where we can start to
      physically connect our devices and send data, but all sending, receiving,
      buffer management, connection synchronisation, and so on, must be managed
-     by the application itself at a level below conventional assembly language,
+     by the application itself rather than being managed by the network stack.
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I don't think this rephrasing reflects the original meaning. I think I'd rephrase to "by the application itself using low-level instructions sets, including quantum assembly (QASM) languages, rather than being managed by a network stack".

File: draft-irtf-qirg-principles-09-rc.xml [hunk 7]
```diff
@@ -308,7 +313,8 @@
 
        <t>a |0> + b |1>,</t>
 
-       <t>where |X> is Dirac's ket notation for a quantum state, here the
+       <t>where |X> is Dirac's ket notation for a quantum state (the value that a 
+       qubit holds), here the
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:318]: Quantum state is more than that, but it's a good point, there probably needs to be some text that explains what a quantum state is and how it is different and relates to a qubit.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 8]
```diff
@@ -534,14 +546,16 @@
...
+         quantum state, error detection and correction is more involved.</t>
 
          <t>To describe the quality of a quantum state, a physical quantity
          called fidelity is used <xref target="NielsenChuang" />. Fidelity takes a
          value between 0 and 1 -- higher is better, and less than 0.5 means the
          state is unusable. It measures how close a quantum state is to the
-         state we have tried to create. It expresses the probability that one
-         state will pass a test to identify as the other. Fidelity is an
+         state we have tried to create. It expresses the probability measuring
+         the actual state yields the same result as measuring the desired target
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: Perhaps, "It expresses the probability that the state will behave exactly the same as our desired state"? Yielding the same result is not necessarily correct I think as that will depend on the measurement.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 9]
```diff
@@ -555,7 +569,8 @@
          its best to deliver it. A higher fidelity can be achieved by either
          having hardware produce states of better fidelity (sometimes one can
          sacrifice rate for higher fidelity) or by employing quantum error
-         detection and correction mechanisms.</t>
+         detection and correction mechanisms (see <xref target="VanMeterBook" />
+         chapter 11).</t>
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: Should probably also add a reference to the generations paper "muhlaridan" I think...

File: draft-irtf-qirg-principles-09-rc.xml [hunk 10]
```diff
@@ -573,7 +588,9 @@
          demands on both resources (physical qubits needed) and their initial
          fidelity. Implementation is very challenging and quantum error
          correction is not expected to be used until later generations of
-         quantum networks.</t>
+         quantum networks. Until then, quantum networks rely on entanglement
+         swapping (<xref target="es" />) and teleportation
+         (<xref target="teleportation" />).</t>
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:594]: Can make a reference to the generations paper again and also refer to the section where the generations are explained.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 11]
```diff
@@ -794,7 +811,7 @@
          is called entanglement swapping. Quantum nodes that create
          long-distance entangled pairs via entanglement swapping are called
          quantum repeaters in academic literature <xref target="Briegel98" />
-         and we will use the same terminology in this memo.</t>
+         and we will use the same terminology in this draft.</t>
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:815]: Is this normal/preferred? It won't be a draft after it's published, no?

File: draft-irtf-qirg-principles-09-rc.xml [hunk 12]
```diff
@@ -1009,7 +1026,9 @@
            networks, we do not need to care about the order in which the Bell
            pairs were generated since they are undirected. This distinction
            makes control algorithms and optimisation of quantum networks
-           different from classical ones. Note that third generation quantum
+           different from classical ones, in the sense that they are more akin
+           to a stateful distributed computation than stateless packet-by-packet
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: I'd just say they are necessarily "stateful" without calling it "stateful distributed computation. Whilst I know some like this phrasing, I know people who will disagree with it (and I'm neutral on this). The document is meant to be a consensus so I'd avoid it.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 13]
```diff
@@ -1275,27 +1296,27 @@ CC - classical channel (can consist of many classical links)
...
 
-         <t>For example, architectures based on optical elements and atomic
-         ensembles <xref target="Sangouard11" /> are very efficient at
+         <t>For example, hardware based on optical elements and atomic
+         ensembles <xref target="Sangouard11" /> is very efficient at
          generating entanglement, but provide little control over the qubits
-         once the pair is generated. On the other hand, nitrogen-vacancy
-         architectures <xref target="Hensen15" /> offer a much greater degree
+         once the pair is generated. On the other hand, nitrogen-vacancy based
+         hardware <xref target="Hensen15" /> offers a much greater degree
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:1314]: We can also add "trapped ions [REF: https://www.nature.com/articles/nature06118]" reference for better coverage.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 14]
```diff
@@ -1369,6 +1390,10 @@ CC - classical channel (can consist of many classical links)
          short memory lifetimes this leads to very tight timing windows to
          build up network-wide connectivity.</t>
 
+         <t>This rate is expected to improve over time as quantum network
+         technology evolves. Rates much higher than 10 Hz are needed to enable
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml:1397]: 10 Hz seems a bit random without a reference.

File: draft-irtf-qirg-principles-09-rc.xml [hunk 15]
```diff
@@ -1602,7 +1635,13 @@ CC - classical channel (can consist of many classical links)
...
            The key service that a quantum network provides is the distribution
            of entanglement between the nodes in a network. All distributed
-           quantum applications are built on top of this key resource. Bell
+           quantum applications are built on top of this key resource. 
+           Applications such as clustered quantum computing, distributed quantum
+           computing, distributed quantum sensing networks, and certain kinds of
+           quantum secure networks require the distribution of quantum state;
+           the distribution of classical state alone is not sufficient.
+           Currently, the only feasible way of reliably distributing quantum state
+           is using teleportation, which requires distribution of entanglement.
```
Comment (Wojtek242) [inline draft-irtf-qirg-principles-09-rc.xml]: That's not strictly correct. Teleportation can indeed do it, but if the state you need can be constructed from Bell states then you don't need teleportation. I would rephrase this a bit.

I would state that "...quantum applications consume entanglement like a resource..." and then state that this fundamental resource can be either used directly in an application, e.g. QKD which simply measures the entangled qubits to obtain a secret key, used to build more complex abstraction and operations, e.g. distributed CNOT gates (https://www.quantum-network.com/applications/distributed-cnot/),  or to directly teleport quantum information.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 2 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:225]: No need to act on this, leaving this for the QIRG mailing list.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 3 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Should also add references to these as well. If you feel like it, you can have a look at the intro in https://dl.acm.org/doi/pdf/10.1145/3386367.3431293 and add them. Feel free, to leave this alone as well and I'll add them later.

**Review by Wojtek242 — CHANGES_REQUESTED** [review]

**Comment by Wojtek242** [comment]
See also e-mail on QIRG mailing list: https://mailarchive.ietf.org/arch/msg/qirg/TNPvzu90ImcvvCXhT__clK6xbSc/

File: draft-irtf-qirg-principles-09-rc.xml (hunk 1 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Done: changed Internet to internet (lower case i)

File: draft-irtf-qirg-principles-09-rc.xml (hunk 2 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:225]: Interesting paper; thanks for pointing to it. I can see how the approach described in this paper allows for a rich topology without using memories or swapping, but I do not see how it overcomes the distance limitations of existing QKD (~ 120 km for BB84, ~ 600 km for TF).  The "long distances" referred to in this paper (start of page 4) are only up to 12.6 km.  Anyway... since you said "no need to act on this, leaving it to the mailing list" I am marking the comment as resolved.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 3 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Added "blind quantum computing" and "clock synchronization". I will leave it to you to add the references.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 4 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Did you notice that I added a reference to [Pompili21] (https://arxiv.org/pdf/2102.04471.pdf).  It seems to me that this goes beyond the "prepare and measure" stage.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 5 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: I have removed the references to Castelvecchi and Wehner.  Note that this contradicts your previous comment where you suggested changing the sentence to "The first fully quantum networks capable of transmitting and manipulating distributed quantum information ("prepare and measure" stage in [WEHNER18]) via an intermediate quantum node have recently been realized" which *does* include the Wehner reference.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 11 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:815]: I'm with you, Woj. But Marie-Jose's response was firm on this. Leaving "draft".

File: draft-irtf-qirg-principles-09-rc.xml (hunk 12 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Updated according to your suggestion.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 13 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: No objection to adding the reference, but what do you suggest we say single-atom qubits to contrast them with atomic ensembles and NV-center?

File: draft-irtf-qirg-principles-09-rc.xml (hunk 14 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1397]: I copied the 10 Hz from the paragraph right above it, which was already there and which does have a reference. I double checked, and the reference actually has the number 10Hz in it (figures 17 through 22 and table 3 all have throughputs peaking below 10 Hz)

File: draft-irtf-qirg-principles-09-rc.xml (hunk 15 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Updated the sentence as you suggested.  Also added the reference to the QNE distributed CNOT page (as well as another reference to the QNE teleportation page).

File: draft-irtf-qirg-principles-09-rc.xml (hunk 3 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Thanks, I'll add them after merging your PR

File: draft-irtf-qirg-principles-09-rc.xml (hunk 4 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Yes, I've seen. But I was just trying to define "fully quantum" network without mentioning entangled states and I think that defining it as "prepare and measure" stage (which defines minimum end-node capabilities) via an intermediate node (so that it's not a p2p link and actually some sort of network) is suitable. And yes, the Delft network is quantum memory stage anyway, but I don't think we have to draw the line that high for the purposes of this draft.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 5 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: Haha yes. I meant kick out the ones at the end that were placed there originally as a reference relative to "imminently realized" to being relative to the mention of quantum network stages.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 11 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:815]: Ack

File: draft-irtf-qirg-principles-09-rc.xml (hunk 13 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: Single-atom qubits is too broad as that could also be neutral atoms.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 14 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1397]: Ah yes, that makes sense. Silly GitHub hiding the paragraph above.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 4 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: I took the easy way out: mention neither entanglement nor prepare-and-measurement at this early stage in the document:

"The first fully quantum networks capable of sending, receiving, and manipulating distributed quantum information have recently been realized [Pompili21]."

Pushed another commit with this change.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 13 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: Yeah, sorry typo. I meant to say "ion-trap" instead of"single-atom". Anyway, can you suggest some specific language to add?

File: draft-irtf-qirg-principles-09-rc.xml (hunk 6 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Restored the language that you suggested. All added a reference to the OpenQASM 3.0 paper.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 8 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Don't you think this may be splitting the hair too finely for a classical network engineer and making the explanation less clear rather than more clear the process?  I thought it was rather obvious from the context that we are doing the same measurement on the desired state and the target state... Anyway, if you feel strongly about it, I can change the language to what you suggested.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 9 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Happy to add it. However, Googling "muhlaridan generations" (or variations thereof) did not yield anything.  Do you have a link to the paper?

File: draft-irtf-qirg-principles-09-rc.xml (hunk 10 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:594]: Ditto - Do you have a link for the paper that you have in mind? I can add it (and the reference to the section).

File: draft-irtf-qirg-principles-09-rc.xml (hunk 9 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: It's already in this draft as "Mural16". Sorry, I just spelled it terribly bad.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 8 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml]: I think that at the same time it is important to be correct though. I do think that some variation "It expresses the probability that the state will behave exactly the same as our desired state" or "It expresses the probability that the state will **mimic the behaviour of the** desired state" is a good balance between the original phrase (which was paraphrased from wikipedia) and your suggestion which is not correct in all circumstances.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 13 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: On the other hand, nitrogen-vacancy <xref target="Hensen15" /> or trapped ion <NEW REF> based hardware offer a much greater degree...

Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: I wouldn't bother contrasting, just adding to the list so that it's not so centered around NVs

File: draft-irtf-qirg-principles-09-rc.xml (hunk 7 above)
Comment (Wojtek242) [reply draft-irtf-qirg-principles-09-rc.xml:318]: On my TODO list

File: draft-irtf-qirg-principles-09-rc.xml (hunk 8 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: I changed the description of fidelity to the first suggestion that you provided.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 9 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml]: Added reference.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 10 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:594]: Added ref to Mural16 fig 2 here.

File: draft-irtf-qirg-principles-09-rc.xml (hunk 13 above)
Comment (brunorijsman) [reply draft-irtf-qirg-principles-09-rc.xml:1314]: Added mention of ion trap, with reference to Moehring07, without specific pros and cons.

**Comment by brunorijsman** [comment]
Okay, I think I have resolved all comments. We still need to run xml2rfc to generate the new text.

**Comment by Wojtek242** [comment]
Awesome, thanks!

**Comment by Wojtek242** [comment]
Can you run xml2rfc to regenerate the text file. Then I'll just do a "squash and merge" and the update will be all in one commit.

**Comment by brunorijsman** [comment]
I ran xml2rfc, produced the .txt files, and fixed some issues that xml2rfc found in the xml file

Go ahead, squash and merge
